dali-mqtt-daemon.py

```diff
@@ -172,22 +188,44 @@ def main_loop(driver, max_lamps, mqtt_server, mqtt_port, ha_prefix):
               }
 
     try:
-        with open(args.config, 'r') as stream:
-            logger.debug("Loading configuration from <%s>", args.config)
-            config = yaml.load(stream)
-
-        MQTT_BASE_TOPIC = config["mqtt_base_topic"]
-
-        main_loop(config["dali_driver"], config["dali_lamps"], config["mqtt_server"], config["mqtt_port"], config["ha_discover_prefix"])
+        config = load_config_file(args.config)
+
+        driver_object = None
+        driver = args.dali_driver
+        logger.debug("Using <%s> driver", driver)
+        if driver == HASSEB:
+            from dali.driver.hasseb import SyncHassebDALIUSBDriver 
+            driver_object = SyncHassebDALIUSBDriver()
+        elif driver == TRIDONIC:
+            from dali.driver.tridonic import SyncTridonicDALIUSBDriver
+            driver_object = SyncTridonicDALIUSBDriver()
+        elif driver == DALI_SERVER:
+            from dali.driver.daliserver import DaliServer
+            driver_object = DaliServer("localhost", 55825)    
+
+        mqqtc = create_mqtt_client(driver_object, config["dali_lamps"], config["mqtt_server"], config["mqtt_port"], config["mqtt_base_topic"], config["ha_discover_prefix"])
+        mqqtc.loop_start()
+
+        watchdog_observer = Observer()
+        watchdog_event_handler = ConfigFileSystemEventHandler(mqqtc, config, args.config, driver_object, loop)
+        watchdog_observer.schedule(watchdog_event_handler, args.config)
+        watchdog_observer.start()
     except FileNotFoundError as e:
-        logger.info("Configuration file %s created, please reload daemon", args.config)
-    except KeyError as e:
-        missing_key = e.args[0]
-        config[missing_key] = args.__dict__[missing_key]
-        logger.info("Configuration file updated, please reload daemon")
```

Comment:
This check was dropped, why ?

Reply:
Well, after adding reload on-fly feature I was think that it doesn't have sense, but probably I misunderstand it so I restored it  with changed message, it seems to me that before message wasn't inform about problem with missing key

Reply:
We can drop the "please reload daemon" part, but the intention is to notify that the Configuration file has been updated by the daemon (e.g. user should backup the new file)

Reply:
Ok, changed